src/pytorch_lightning/strategies/launchers/multiprocessing.py

```diff
@@ -97,7 +98,9 @@ def launch(self, function: Callable, *args: Any, trainer: Optional["pl.Trainer"]
         # The default cluster environment in Lightning chooses a random free port number
         # This needs to be done in the main process here before starting processes to ensure each rank will connect
         # through the same port
-        os.environ["MASTER_PORT"] = str(self._strategy.cluster_environment.main_port)
+        cluster_environment = self._strategy.cluster_environment or LightningEnvironment()
```

Comment:
this would actually never happen with the Trainer/Lite. 

`_MultiProcessingLauncher` is not meant to be used with arbitrary things as input. It assumes that the strategy given here is properly set up by the Connector. It is why these launchers are not public API.

Reply:
I think it's good to allow using the launchers like this. Makes launching processes super easy and simple. I am going to move this into a separate PR though.

Reply:
Synced in a call: dropping this change, hopefully we can do this one day with lazy init.